src/main/webapp/app/exam/participate/summary/exam-result-summary.component.ts

```diff
@@ -253,5 +324,128 @@ export class ExamResultSummaryComponent implements OnInit {
+                });
+                continue;
+            }
+
+            const { textColorClass, resultIconClass } = this.getTextColorAndIconClassByExercise(exercise);
+
+            exerciseInfos[exercise.id] = {
+                icon: getIcon(exercise.type),
+                isCollapsed: false,
+                achievedPoints: this.getPointsByExerciseIdFromExam(exercise.id, studentExamWithGrade),
+                achievedPercentage: this.getAchievedPercentageByExerciseId(exercise.id),
+                colorClass: textColorClass,
+                resultIconClass: resultIconClass,
+
+                submission: this.getSubmissionForExercise(exercise),
+                participation: this.getParticipationForExercise(exercise),
+                displayExampleSolution: false,
+            };
+        }
+        return exerciseInfos;
+    }
+
+    private getPointsByExerciseIdFromExam(exerciseId: number, studentExamWithGrade?: StudentExamWithGradeDTO): number | undefined {
+        if (!studentExamWithGrade) {
+            return undefined;
+        }
+
+        for (const achievedPointsPerExerciseKey in this.studentExamGradeInfoDTO?.achievedPointsPerExercise) {
+            if (Number(achievedPointsPerExerciseKey) === exerciseId) {
+                return this.studentExamGradeInfoDTO.achievedPointsPerExercise[achievedPointsPerExerciseKey];
+            }
+        }
+
+        return undefined;
+    }
+
+    private getExerciseResultByExerciseId(exerciseId?: number): ExerciseResult | undefined {
+        if (exerciseId === undefined) {
+            return undefined;
+        }
+
+        const exerciseGroupResultMapping = this.studentExamGradeInfoDTO?.studentResult?.exerciseGroupIdToExerciseResult;
+        let exerciseResult = undefined;
+
+        for (const key in exerciseGroupResultMapping) {
+            if (key in exerciseGroupResultMapping && exerciseGroupResultMapping[key].exerciseId === exerciseId) {
+                exerciseResult = exerciseGroupResultMapping[key];
+                break;
+            }
+        }
+
+        return exerciseResult;
+    }
+
+    toggleShowSampleSolution(exerciseId?: number) {
+        if (exerciseId === undefined) {
+            const errorMessage = 'Cannot show sample solution because exercise id is undefined';
+            console.error(errorMessage, exerciseId);
+            captureException(new Error(errorMessage), {
+                extra: {
+                    exerciseId,
+                },
+            });
+
+            return;
+        }
+
+        this.exerciseInfos[exerciseId].displayExampleSolution = !this.exerciseInfos[exerciseId].displayExampleSolution;
+    }
+
+    getAchievedPercentageByExerciseId(exerciseId?: number): number | undefined {
+        const result = this.getExerciseResultByExerciseId(exerciseId);
+        if (result === undefined) {
+            return undefined;
+        }
+
+        const course = this.studentExamGradeInfoDTO.studentExam?.exam?.course;
```

Comment:
`course` is only used in the next if statement and can thus be moved into it.

Reply:
It is also used in the subsequent if statement, and think that it improves the readability and reduces the error proneness of just changing it within one if statement